griptape/tools/google_sheets/tool.py

```diff
@@ -0,0 +1,400 @@
+                            fileId=sheet_id,
+                            mimeType=export_mime_mapping[mime_type],
+                        )
+                        downloaded_files.append(BlobArtifact(request.execute()))
+                    else:
+                        logging.error(
+                            f"error finding spreadsheet at path: {file_path}"
+                        )
+
+            except HttpError as e:
+                logging.error(e)
+
+            except MalformedError:
+                logging.error(
+                    "error downloading spreadsheet due to malformed credentials"
+                )
+
+        return ListArtifact(downloaded_files)
+
+    @activity(
+        config={
+            "description": "Uploads a spreadsheet and converts it to a Google Sheets format",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_name",
+                        description="The name of the file to be uploaded",
+                    ): str,
+                    Literal(
+                        "file_path",
+                        description="The local path to the file to be uploaded",
+                    ): str,
+                    Literal(
+                        "file_type",
+                        description="The type of the file being uploaded, e.g., 'csv' or 'excel'",
+                    ): str,
+                }
+            ),
+        }
+    )
+    def upload_spreadsheet(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        from googleapiclient.http import MediaFileUpload
+
+        mime_mapping = {
+            "csv": "text/csv",
+            "excel": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+        }
+
+        file_type = params["file_type"]
+        if file_type not in mime_mapping:
+            logging.error(f"Unsupported file type '{file_type}' provided.")
+            return ErrorArtifact(
+                f"Unsupported file type '{file_type}'. Please provide either 'csv' or 'excel'."
+            )
+
+        try:
+            service = self._build_client(
+                scopes=self.DRIVE_UPLOAD_SCOPES,
+                service_name="drive",
+                version="v3",
+                owner_email=self.owner_email,
+            )
+
+            file_metadata = {
+                "name": params["file_name"],
+                "mimeType": "application/vnd.google-apps.spreadsheet",
+            }
+            media = MediaFileUpload(
+                params["file_path"], mimetype=mime_mapping[params["file_type"]]
+            )
+            file = (
+                service.files()
+                .create(body=file_metadata, media_body=media, fields="id")
+                .execute()
+            )
+
+            return InfoArtifact(
+                f'file was successfully uploaded and converted to a Google Sheet with ID {file.get("id")}.'
+            )
+
+        except Exception as e:
```

Comment:
Can we standardize the exception handling? Some activities catch `Execption`, others `HttpError` + `MalformedError`. You know the errors best, what is the best combination that we can apply to all the activities?

Reply:
I think we should have Http and Malformed for all as well as a common Exception for all other ones. It would be the safest option.